### ISSUE TITLE
Wiki feedback

### ISSUE THREAD

**Issue by RunDevelopment** [muslll/neosr#21]
Hi! I read through the [Configuration Walkthrough](https://github.com/muslll/neosr/wiki/Configuration-Walkthrough) wiki page and have a few questions. Could you answer these questions and maybe add the answers to the wiki?

> The compile option is experimental. This option enables pytorch's new torch.compile(), which can speed up training. As of this writing, pytorch 2.0.0 does not have support for compile using Python 3.11. Pytorch version 2.1 is expected to fix this, as well as better Windows support.

NeoSR now requires 2.1, so is the last part still relevant?

> The manual_seed option enables deterministic training. If you're not doing precise comparisons, this option should be commented out, otherwise training performance will decrease.

What are "precise comparisons"? Or rather, when should this be used?

> The gt_size is one of the most important options you have to change. It sets the size that each image will be cropped before being sent to the network.

How will it be cropped/which portion will be used? Will it use random offsets?

**Comment by Phhofm** [comment]
Just saw this issue and thought id comment what I think, maybe it is helpful, musl can of course still comment officially later.

1. This would probably need testing. But I had out of curiosity used compiled on for triaining compact and it worked and output looked normal. So it worked on my system, yes. But this was only a fast compact test and speedup was small (but there was a speedup). Bigger archs would probably benefit more of a speedup, but I have not tested yet. Not sure if musl can use this option with his gpu, I think community help to test this out would be needed. All I can say is it worked on my ubuntu system when I tried it out once, but only tested it for around 5k or something iters, so real tests would be needed.

2. When comparing arches with eachother. Meaning this option would be used if the goal is not to get the best output for a model, but to compare arches with eachother. This should minimize randomization that happens during training as far as I understand with my very limited understanding.
What I mean is: the goal were training multiple models with the exact same setting. For example a dat2 model with gt_size 128, batch size 6, paired dataset etc. And then the exact same setting for a compact model for example, with gt_size 128, batch size 6, same paired dataset etc. In a normal case, I would crank up gt_size to 512 and batch size also a bit more for the compact model, since it uses way less vram, so I could get better results, since normally my goal is to get the 'best' model I can train for something. But like I described, the goal here would be to compare different arches (for example CRAFT with OmniSR) with the exact same settings. This option tries to eliminate randomization that happens during training so the comparison is more fair (therefore 'precise comparison'). (like with sd generation, comparing different models with same input prompt but different seeds, even the same model would create a very different image with a different seed with same prompt as input. Not activating this option is like using random seed for each comparison with the same prompt, so these would be unfair comparisons. So the goal were same input prompt, same seed, for a comparison, something like that. I have very limited understanding of sd image generation though so not sure if this comparison checks out. Im just trying to convey the use case here).
-> use it if the foal is fair comparison of arches (swinir-s vs sformer-small etc). Dont use it if the goal is simply to train a good model (faster training).

Have no idea about question number 3 sry. Just thought Id provide my thoughts on the first two, but then again I lack understanding, musl can still answer correctly, sorry for chiming in, I simply thought some of it might help after seeing this issue.

**Comment by muslll** [comment]
Hi! Thanks for answering Philip.

1. I forgot to update the wiki. Thanks for pointing it out.
2. By "precise comparisons" I mean reproducible ones. Using `manual_seed` will not only set the seeds for the pseudo-RNGs (for example, making pytorch dataloader fetch the exact same images in the same order), it will also turn on [pytorch `use_deterministic_algorithms`](https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html), which aims at removing non-deterministic behavior from pytorch. Using this is very important in case you're doing experiments, as it will eliminate variables like randomness and heuristics that could interfere with the validity of the test.
3. Patchs are cropped based on the `gt_size` value (in pixels). It crops a random part of the GT image, and the same random part of the LQ (with size `gt_size` divided by `scale`). A new patch of each image is generated at each epoch.

Answering to Philip:
> I think community help to test this out would be needed

I would really appreciate more tests on torch Dynamo `compile()`. I indeed can't run this test because my gpu is not supported by Triton.

> which would flaw these metric comparisons a bit, which he tried to minimize.

Yes, BasicSR code is not as strict in that regard. It doesn't set all seeds, and it doesn't enable deterministic pytorch. To see what exactly neosr does different, see [L172 of options.py](https://github.com/muslll/neosr/blob/master/neosr/utils/options.py#L172) and [L11 of misc.py](https://github.com/muslll/neosr/blob/master/neosr/utils/misc.py#L11) (I also switched to latest RNG apis of numpy and pytorch).
